material-ui/material-ui.d.ts

```diff
@@ -1131,7 +1136,8 @@ declare namespace __MaterialUI {
         export class TableHeader extends React.Component<TableHeaderProps, {}> {
         }
 
-        interface TableHeaderColumnProps extends React.Props<TableHeaderColumn> {
+        interface TableHeaderColumnProps extends HasStyleAndClassNameProps<TableHeaderColumn> {
```

Comment:
Based on the `render()` method in the material-ui source code, since `th` is passed all `other`  properties, use `React.HTMLAttributesBase<TableHeaderColumn>` instead of `HasStyleAndClassNameProps`.